internal/bgp/bgp.go

```diff
@@ -665,3 +679,30 @@ func buildTCPMD5Sig(addr net.IP, key string) tcpmd5sig {
 
 	return t
 }
+
+// localAddressExists returns true if the provided address exists locally on
+// any of the host's network interfaces. The function doesn't return errors and
+// instead returns false if an error has occurred while looking for the
+// address. This function is safe to call if addr is nil or malformed.
+func localAddressExists(addr string) bool {
```

Comment:
Just curious: why use this function to see if the srcIP is present instead of trying to use it directly? The error is not clear if we just use it? Or it may allow sending some weird packages over the network to a user that can just modify metalLB config?

Just to understand the reasoning. Seems legit.

Reply:
With your comment [here][link], I doubt if we want to check something at all.  I'm sure you add it for a reason, but can you please share it? :)


[link]: https://github.com/metallb/metallb/pull/672#discussion_r458050524